pkg/operator/operator.go

```diff
@@ -172,22 +164,43 @@ func (o *Operator) watchTPR(watchVersion string) error {
 
 }
 
+func (o *Operator) startCluster(cluster *Cluster) {
+	if o.devicesInUse && cluster.Spec.UseAllDevices {
```

Comment:
at restart, devicesInUse under the covers actually will only be applied on the first start of the cluster. The OSD daemonset gets the original value and it won't be updated through the lifetime of the daemonset.